lib/src/components/NxThreatIndicator/__tests__/NxThreatIndicator.test.tsx

```diff
@@ -4,116 +4,115 @@
-        .toMatchSelector('.nx-threat-indicator--severe');
-    expect(getShallowComponent({ policyThreatLevel: 9 }).find('.nx-threat-indicator'))
-        .toMatchSelector('.nx-threat-indicator--critical');
-    expect(getShallowComponent({ policyThreatLevel: 9 }).find('.nx-threat-indicator'))
-        .not.toMatchSelector('.nx-threat-indicator--unspecified');
-  });
-
-  it('sets the modifier class using the threatLevelCategory if both props are provided', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
-    const icon = component.find(NxFontAwesomeIcon);
-
-    expect(icon).toMatchSelector('.nx-threat-indicator--low');
-    expect(icon).not.toMatchSelector('.nx-threat-indicator--critical');
-    expect(icon).not.toMatchSelector('.nx-threat-indicator--unspecified');
+    await user.hover(threatIndicator);
+    const tooltip = await screen.findByRole('tooltip');
+    expect(tooltip).toHaveTextContent('Low');
   });
 
   it('adds aria attrs to help the icon show up for screen readers', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
+    const el = quickRender({ policyThreatLevel: 9, threatLevelCategory: 'low' }),
+        threatIndicator = el.getByRole('img', { hidden: true, queryFallbacks: true })!;
 
-    expect(component.find(NxFontAwesomeIcon)).toHaveProp('aria-label', 'threat level low');
-    expect(component.find(NxFontAwesomeIcon)).not.toHaveProp('aria-hidden');
+    expect(threatIndicator).toHaveAttribute('aria-label', 'threat level low');
+    expect(threatIndicator).not.toHaveAttribute('aria-hidden', true);
   });
 
-  it('should have the correct default tooltip title', function() {
-    const none = getShallowComponent({ threatLevelCategory: 'none' }).find(NxToolip);
-    const low = getShallowComponent({ threatLevelCategory: 'low' }).find(NxToolip);
-    const moderate = getShallowComponent({ threatLevelCategory: 'moderate' }).find(NxToolip);
-    const severe = getShallowComponent({ threatLevelCategory: 'severe' }).find(NxToolip);
-    const critical = getShallowComponent({ threatLevelCategory: 'critical' }).find(NxToolip);
-
-    expect(none).toHaveProp('title', 'None');
-    expect(low).toHaveProp('title', 'Low');
-    expect(moderate).toHaveProp('title', 'Moderate');
-    expect(severe).toHaveProp('title', 'Severe');
-    expect(critical).toHaveProp('title', 'Critical');
+  describe('should have default tooltips', function() {
+    describe('when policyThreatLevel prop is provided', function() {
+      allThreatLevelNumbers.forEach(threatNumber => {
+        it(`for policy threat level ${threatNumber}`, async function() {
+          const el = quickRender({ policyThreatLevel: threatNumber })!,
+              user = userEvent.setup(),
+              category = categoryByPolicyThreatLevel[threatNumber],
+              categoryTitleCase = category[0].toUpperCase() + category.slice(1),
```

Comment:
This is too programmatic, and too reliant on the same data structures that drive the implementation.  As a test, this should just hardcode the relationships between the expected inputs and outputs.  e.g.
```
expect(getTooltipTextForProps({ policyThreatLevel: 1 })).toBe('Low');
```
and so on.  And you'd have to write the `getTooltipTextForProps` function.

Reply:
> This is too programmatic, and too reliant on the same data structures that drive the implementation. As a test, this should just hardcode the relationships between the expected inputs and outputs. e.g.
> 
> ```
> expect(getTooltipTextForProps({ policyThreatLevel: 1 })).toBe('Low');
> ```
> 
> and so on. And you'd have to write the `getTooltipTextForProps` function.

Changed here: 8900977e452b45b41a5a7c46e7a9b8015a421cae